lib/pbench/cli/server/report.py

```diff
@@ -161,32 +68,29 @@ def report_archive(tree: CacheManager):
     watcher.update("inspecting archive")
     tarball_count = len(tree.datasets)
     tarball_size = 0
-    smallest_tarball = GINORMOUS
-    smallest_tarball_name = None
-    biggest_tarball = 0
-    biggest_tarball_name = None
+    tcomp = Comparator("tarball")
+    usage = shutil.disk_usage(tree.archive_root)
 
     for tarball in tree.datasets.values():
         watcher.update(f"({tarball_count}) archive {tarball.name}")
         size = tarball.tarball_path.stat().st_size
         tarball_size += size
-        if size < smallest_tarball:
-            smallest_tarball = size
-            smallest_tarball_name = tarball.name
-        if size > biggest_tarball:
-            biggest_tarball = size
-            biggest_tarball_name = tarball.name
+        tcomp.add(size, tarball.name)
     click.echo("Archive report:")
+    click.echo(
+        f"  ARCHIVE ({tree.archive_root}): {humanize.naturalsize(usage.total)}: {humanize.naturalsize(usage.used)} "
+        f"used, {humanize.naturalsize(usage.free)}"
+    )
     click.echo(
         f"  {tarball_count:,d} tarballs consuming {humanize.naturalsize(tarball_size)}"
     )
     click.echo(
-        f"  The smallest tarball is {humanize.naturalsize(smallest_tarball)}, "
-        f"{smallest_tarball_name}"
+        f"  The smallest tarball is {humanize.naturalsize(tcomp.min)}, "
+        f"{tcomp.min_name}"
     )
```

Comment:
Yeah, I did consider consolidating the formatting in `Comparator`, but the formats are (and I think need to be) enough different that it would get messier than seemed justified by any real value.

And, yeah, the dangling name formatter looks a bit odd, but all I did in this round was change the text to reference the object property and I don't really like the reasonable alternatives, either. While I could switch to `format`, I would be displeased with switching back and forth: nor would I want to change to `format` for all of the output as I really prefer the f-string style. Ultimately, I think I'll leave it as-is.